nixos/modules/services/misc/n8n.nix

```diff
@@ -0,0 +1,433 @@
+            default = "";
+            description = "The password of the basic auth user (Beware when you put SECRETS in the store, it is world readable)";
+          };
+
+          hash = mkOption {
+            type = types.bool;
+            default = false;
+            description = "If password for basic auth is hashed";
+          };
+        };
+
+        jwtAuth = {
+          active = mkEnableOption "If jwt auth should be activated for editor and REST-API";
+
+          jwtHeader = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header containing a signed JWT";
+          };
+
+          jwtHeaderValuePrefix = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header value prefix to strip (optional)";
+          };
+
+          jwksUri = mkOption {
+            type = types.str;
+            default = "";
+            description = "The URI to fetch JWK Set for JWT authentication";
+          };
+
+          jwtIssuer = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT issuer to expect (optional)";
+          };
+
+          jwtNamespace = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT namespace to expect (optional)";
+          };
+
+          jwtAllowedTenantKey = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant key name to inspect within JWT namespace (optional)";
+          };
+
+          jwtAllowedTenant = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant to allow (optional)";
+          };
+        };
+      };
+
+      endpoints = {
+        rest = mkOption {
+          type = types.str;
+          default = "rest";
+          description = "Path for rest endpoint";
+        };
+
+        webhook = mkOption {
+          type = types.str;
+          default = "webhook";
+          description = "Path for webhook endpoint";
+        };
+
+        webhookTest = mkOption {
+          type = types.str;
+          default = "webhook-test";
+          description = "Path for test-webhook endpoint";
+        };
+      };
+
+      externalHookFiles = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Files containing external hooks. Multiple files can be separated by colon (\":\")";
+      };
+
+      nodes = {
+        exclude = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          description = "Nodes not to load";
+        };
+
+        errorTriggerType = mkOption {
+          type = types.str;
+          default = "n8n-nodes-base.errorTrigger";
+          description = "Node Type to use as Error Trigger";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.n8n = {
```

Comment:
I noticed the upstream repo doesn't have a `systemd` unit file. They might benefit from you submitting a PR to add one.